### PR TITLE
[SPARK-45908][Python] Add support for writing empty DataFrames to parquet with partitions

### PR DESCRIPTION
**Pull request by ti1uan**
### What changes were proposed in this pull request?

This change introduces a new functionality in the `parquet` method of the DataFrameWriter class to handle the writing of empty DataFrames to Parquet files, particularly when using partitioning. Previously, writing an empty DataFrame with partitions specified did not create any output in the target directory, which could lead to issues in subsequent jobs expecting files with the defined schema. Now, `parquet` method will check if the DataFrame is empty and if partitions are specified. If both conditions are true, private method `_write_empty_partition` is called to handle the empty DataFrame write operation.

### Why are the changes needed?

This change addresses the issue reported in [SPARK-45908](https://issues.apache.org/jira/browse/SPARK-45908) regarding the handling of empty DataFrames with partitions in PySpark's Parquet writing functionality.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually tested.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -1936,7 +1936,23 @@ def parquet(
...
+        ...
+    
+    @overload
+    def _write_empty_partition(self, path: str, partitionBy: List[str]) -> None:
+        ...
+
+    def _write_empty_partition(self, path: str, partitionBy: Union[str, List[str]]) -> None:
+            temp_row_data = {col: None for col in partitionBy} if isinstance(partitionBy, list) else {partitionBy: None}
+            temp_df = self._df.sparkSession.createDataFrame([Row(**temp_row_data)])
+            temp_df.write.mode("overwrite").partitionBy(partitionBy).parquet(path)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:1955]: Does this work with Scala API too?

Comment (maintian) [reply python/pyspark/sql/readwriter.py:1955]: Thanks for your review. No, this change only applies for PySpark, Scala API has no change with it. Do you think we should support it for Scala API?

Comment (HyukjinKwon) [reply python/pyspark/sql/readwriter.py]: The change has to be made for API parity.

Comment (maintian) [reply python/pyspark/sql/readwriter.py]: I will change this PR to draft and work on changes for Scala API

Comment (maintian) [reply python/pyspark/sql/readwriter.py]: Hi @HyukjinKwon, I'm under the impression that PySpark relies on the underlying Scala API for Parquet operations. If that's correct, would updating the Scala API alone be sufficient to introduce this behavior to both PySpark and Scala API?

**Pull request closed without merging**